pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation_test.go

```diff
@@ -2246,10 +2342,11 @@ func TestShouldInject(t *testing.T) {
 
 			// Need to create a new instance of the webhook to take into account
 			// the config changes.
-			apmInstrumentationWebhook, errInitAPMInstrumentation = NewWebhook(wmeta)
+			UnsetWebhook()
+			webhook, errInitAPMInstrumentation := GetWebhook(wmeta)
 			require.NoError(t, errInitAPMInstrumentation)
 
-			if got := ShouldInject(tt.pod, wmeta); got != tt.want {
+			if got := webhook.isPodEligible(tt.pod); got != tt.want {
```

Comment:
I think we should change the name of the test because now we're testing a different function.

Reply:
Adding a test for both `ShouldInject` and `isPodEligible`